benchopt/utils/files.py

```diff
@@ -25,7 +31,30 @@ def uniquify_results(file_path):
         alternative = parent / f"{stem}_{i}{suffix}"
         warnings.warn(
             f"{file_path} already exists. Saving results to {alternative}"
-            )
+        )
         return alternative
     else:
         return file_path
+
+
+def read_results(file_path):
+    """Read benchmark results as pandas DataFrame."""
+    with open(file_path, "r") as f:
+        metadata = {}
+        for line in takewhile(lambda s: s.startswith("#"), f):
+            k, v = line.split(':')
+            metadata[k.strip('#')] = v.strip('\n')
+        df = pd.read_csv(f, comment='#')
+    return df, metadata
+
+
+def write_results(df, file_path):
+    """Write benchmark results to CSV file."""
+    err, tag = _run_shell("git describe --tags --abbrev=0", return_output=True)
+    if err != 0:
+        if DEBUG:
+            print(err, tag)
+        tag = None
```

Comment:
This should be a dedicated function.

Reply:
Maybe in `sys_info.py`.

Reply:
Thinking back, why didn't we put this directly in `sys-info`? this would avoid this change and be directly accessible as the other sys info?

Reply:
yes, we can